### ISSUE TITLE
Mandatory Questionaire

### ISSUE THREAD

**Issue by MacBlub** [vsaase/covid19selfreport#3]
The current system does only show the infection map, but does not invite/incite the users to fill out the questionaire.  

As we want to have a good estimate of the current infection-spread, we are dependent on :
A) Recent reportings of the users

I request the follwing feature:
Mandatory Questionaire by first Leading the visitor to the questionaire page and then if completed leading the user to the visualization.

**Comment by MacBlub** [comment]
What do you want to work on? I would be working on getting the "mandatory" part, working. Do you only handle the updated form, or also the mandatory part?

**Comment by mctraub** [comment]
We started collecting candidate questions here: https://docs.google.com/document/d/15o_RwKcmkYQl8328agtYGyjIp9yKJtugYfnDI9Ipsro/edit?usp=sharing
Next step is to sort them into the **mandatory part** that the user fills in when visiting the page for the first time and the **update part**, that should be filled in on a daily basis (to track the development of the symptoms of users).

**Comment by MacBlub** [comment]
Do you already know, how this could be implemented?

I think a stateless implementation is not possible with this requirement. Meaning we have to handle sessions...

Btw.: Do you want to create a feature branch? There we could update the questionaire accordingly.

**Comment by carolina-ott** [comment]
We developed a first draft of the questionnaire, you can find it at the bottom of the google-doc mentioned here. We stumbled across a couple of questions:
- In which format will the questionanaire be needed later?
- How accurate shoud people fill in their location, just by zip code or by Adress?
- Do people signing up get some kind of account, so they can update their information on their state of health at a later stage?
- Can people only track themselves or others as well, for example, I would like to give information on my grandmother, who can not give the information on her own

Can we discuss those question?

**Comment by MacBlub** [comment]
Yes, that would be a good idea.

1. Regarding Format, we should ask the backend-/model-teams, what Info they actually need.
2. My understanding was ZIP code
3. Currently, I handle stuff via cookies. Meaning you should use the same device for an update. Here the questions: We could handle Accounts using O2Auth? Though, I cannot esimtate how much of an effort this is...
4. Tracking others is problematic, because this means we have to think, how others can update the report of others. ID, E-Mail, ... ?

**Comment by carolina-ott** [comment]
Some questions have been answered in the Slack Thead as well.

**Comment by MacBlub** [comment]
Need some help here: 
1. Update the questionaire with the correct questions
2. Create a questionaire, where infos can be updated.

**Comment by Skhurana136** [comment]
> Need some help here:
> 
> 1. Update the questionaire with the correct questions
> 2. Create a questionaire, where infos can be updated.

I can give #1 a shot...if someone hasn't done it already

**Comment by Skhurana136** [comment]
My understanding is that the final questionnaire was the "Consolidated Questionnaire" which is very similar to what we had already. If more questions are to be added, please clarify. Also please check German too, I used an online translator for some of the questions... I should have written this in the commit comment.

**Comment by carolina-ott** [comment]
Yes, the last part in the google doc is the questionnaire so far. At the moment the questionnaire is held shortly, so users are not overwelmed with questions. If there are more question that sould be added, let me know. I can check the German as well =)

**Comment by vsaase** [comment]
> > Need some help here:
> > 
> > 1. Update the questionaire with the correct questions
> > 2. Create a questionaire, where infos can be updated.
> 
> I can give #1 a shot...if someone hasn't done it already

thanks, your changes were merged with some minor fixes

**Comment by mctraub** [comment]
One thing is still missing in the questionnaire: The user should be able to indicate how strong each of the symptoms are. If we have this data, we can find out if s/he is already recovering or if it's getting worse.
